### PR TITLE
Weighted regression

### PR DESCRIPTION
**Pull request by QuantumPlumber**
This PR covers numeric dataset weighting for XGBoost and LightGBM, and associated tests. Users can now provide a json dictionary of histogram bin edges and associated weights, which will be used to map the labels to weights. The histogram bins are left edge -- overflows are assigned to the last bin's weight, and underflows to the first bin's weight.

### CODE REVIEW

**Review by paxcema — CHANGES_REQUESTED** [review]
Mostly minor things to comment on. The one big change still needed is the merge with `staging` then ensuring CI installs `lightgbm`, but not making it a default dep just yet.

The rest I went through on my end, and it works just as advertised 🙌

File: requirements.txt [hunk 1]
```diff
@@ -32,3 +32,4 @@ tab-transformer-pytorch >= 0.2.1
 typing-inspect
 six
 regex
+lightgbm
```
[the remark this answers is not in the document]
Comment (QuantumPlumber) [reply requirements.txt]: Merged staging in, which removed the requirements file.

File: tests/unit_tests/mixer/test_lgbm.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
+import unittest
+import numpy as np
+import pandas as pd
+from lightwood.api.types import ProblemDefinition
+from lightwood.api.high_level import json_ai_from_problem, code_from_json_ai, predictor_from_code
+
+np.random.seed(42)
+
+
+class TestBasic(unittest.TestCase):
```
[the remark this answers is not in the document]
Comment (QuantumPlumber) [reply tests/unit_tests/mixer/test_lgbm.py:17]: Added a skip if  `lightgbm` cant be imported

File: tests/unit_tests/mixer/test_xgboost.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
+import unittest
+import numpy as np
+import pandas as pd
+from lightwood.api.types import ProblemDefinition
+from lightwood.api.high_level import json_ai_from_problem, code_from_json_ai, predictor_from_code
+
+np.random.seed(42)
+
+
+class TestBasic(unittest.TestCase):
```
[the remark this answers is not in the document]
Comment (QuantumPlumber) [reply tests/unit_tests/mixer/test_xgboost.py:10]: Woohoo!

**Review by paxcema — APPROVED** [review]
